store/details_v2.go

```diff
@@ -87,6 +87,136 @@ type storeSnapMedia struct {
 	Height int64  `json:"height"`
 }
 
+// storeInfoChannel is the channel description included in info results
+type storeInfoChannel struct {
+	Architecture string `json:"architecture"`
+	Name         string `json:"name"`
+	Risk         string `json:"risk"`
+	Track        string `json:"track"`
+}
+
+// storeInfoChannelSnap is the snap-in-a-channel of which the channel map is made
+type storeInfoChannelSnap struct {
+	storeSnap
+	Channel storeInfoChannel `json:"channel"`
+}
+
+// storeInfo is the result of v2/info calls
+type storeInfo struct {
+	ChannelMap []*storeInfoChannelSnap `json:"channel-map"`
+	Snap       storeSnap               `json:"snap"`
+	Name       string                  `json:"name"`
+	SnapID     string                  `json:"snap-id"`
+}
+
+func infoFromStoreInfo(si *storeInfo) (*snap.Info, error) {
+	if len(si.ChannelMap) == 0 {
```

Comment:
this could be empty afaiu, for a public registered snap without any release yet (it's not the case atm but seems there is no absolute promise that not to change)

Reply:
fixed to return ErrSnapNotFound